src/charm.py

```diff
@@ -75,6 +87,60 @@ def _on_update_status(self, event):
 
         self.unit.status = ActiveStatus()
 
+    def _on_create_user_action(self, event):
+        """Handle the create_user action."""
+        creds = {
+            "username": event.params["username"],
+            "password": event.params["password"],
+            "hostname": "%",
+        }
+
+        try:
+            self.mysql.new_super_user(creds)
```

Comment:
Will this actually delete the previous user?

Reply:
No, this action does not delete previous users.
This action verifies if the username is already taken. If so will show `"User already exists: {credentials['username']}"`